### PR TITLE
Fix underflow in format_imports

### PR DESCRIPTION
**Pull request by pepyakin**
Fixes underflow described in #1116.

### CODE REVIEW

File: src/imports.rs [hunk 1]
```diff
@@ -159,8 +159,15 @@ impl Rewrite for ast::ViewPath {
...
     pub fn format_imports(&mut self, use_items: &[ptr::P<ast::Item>]) {
-        let mut last_pos =
-            use_items.first().map(|p_i| p_i.span.lo - BytePos(1)).unwrap_or(self.last_pos);
+        let mut last_pos = use_items.first()
+            .and_then(|p_i| {
+                if p_i.span.lo.0 > 0 {
+                    Some(p_i.span.lo - BytePos(1))
+                } else {
+                    None
+                }
```
Comment (nrc) [inline src/imports.rs]: you could use `checked_sub` here rather than the `if` check.

**Comment by pepyakin** [comment]
In [previous commit](https://github.com/pepyakin/rustfmt/commit/041f1c7aaa117cdef039e2d5e88cabce5600eccd) I accidentally reformated imports.rs with reorder-imports enabled. Sorry about that.

File: src/imports.rs [hunk 2]
```diff
@@ -159,8 +159,9 @@ impl Rewrite for ast::ViewPath {
 
 impl<'a> FmtVisitor<'a> {
     pub fn format_imports(&mut self, use_items: &[ptr::P<ast::Item>]) {
-        let mut last_pos =
-            use_items.first().map(|p_i| p_i.span.lo - BytePos(1)).unwrap_or(self.last_pos);
+        let mut last_pos = use_items.first()
+            .and_then(|p_i| p_i.span.lo.0.checked_sub(1).map(|span_lo| BytePos(span_lo)))
```
Comment (nrc) [inline src/imports.rs]: Can the `map` go outside the `and_then`? So we have `.first().and_then(...).map(...).unwrap_or(...)`

Comment (pepyakin) [inline src/imports.rs]: Yeah, sure.

**Comment by nrc** [comment]
Thank you!

**Pull request merged**